e2e/rbd.go

```diff
@@ -2060,20 +2060,85 @@ var _ = Describe("RBD", func() {
 			By("validate the functionality of controller", func() {
 				err := deleteResource(rbdExamplePath + "storageclass.yaml")
 				if err != nil {
-					e2elog.Failf("failed to delete storageclass with error %v", err)
+					e2elog.Failf("failed to delete storageclass : %v", err)
 				}
-				err = validateController(f, pvcPath, appPath, rbdExamplePath+"storageclass.yaml")
+				volumeNamePrefix := "test-"
+				err = validateController(f,
+					pvcPath, appPath, rbdExamplePath+"storageclass.yaml",
+					nil,
+					map[string]string{
+						"volumeNamePrefix": volumeNamePrefix,
+					},
+					true)
 				if err != nil {
-					e2elog.Failf("failed to validate controller with error %v", err)
+					e2elog.Failf("failed to validate controller : %v", err)
 				}
 				// validate created backend rbd images
 				validateRBDImageCount(f, 0, defaultRBDPool)
 				err = createRBDStorageClass(f.ClientSet, f, defaultSCName, nil, nil, deletePolicy)
 				if err != nil {
-					e2elog.Failf("failed to create storageclass with error %v", err)
+					e2elog.Failf("failed to create storageclass : %v", err)
 				}
 			})
 
+			By("validate the functionality of controller with various encryptions", func() {
+				err := deleteResource(rbdExamplePath + "storageclass.yaml")
+				if err != nil {
+					e2elog.Failf("failed to delete storageclass : %v", err)
+				}
+				encryptionKMSIDList := []string{
```

Comment:
@nixpanic , All these encryption types are used to 
- create pvc,pv(with retain policy) 
- pv,pvc backed up
- pv,pvc,sc deleted
- pv,pvc,sc recreated (omap generator/rbd controller fills in omap here)
- create app which attachs these pvc
- delete everything 

So for encryption we are depending on Volid, kmsid & owner which is filled by this pr and the above e2e tests all aspects of it. 
Please let me know if I am missing anything

cc @Madhu-1

Reply:
come to think of it, maybe this list should be maintained in `e2e/kms.go` so that it is easy to add more and we won't forget about updating the list when more KMSs are implemented?

Reply:
current E2E is just a hack to verify some content. we need to data of mirrored encrypted PVC for all the encryption type cephcsi supports. we should be able to do all CSI operations after failover and the application should be able to read the existing data and should be able to write new data.